pgrx-sql-entity-graph/src/pg_extern/returning.rs

```diff
@@ -111,62 +102,40 @@ impl Returning {
                     };
 
                     let mut segments = option_inner_path.segments.clone();
-                    let mut found_option = false;
-                    'outer: loop {
-                        for segment in &segments {
-                            let ident_string = segment.ident.to_string();
-                            match ident_string.as_str() {
-                                "Option" => match &segment.arguments {
-                                    PathArguments::AngleBracketed(bracketed) => {
-                                        match bracketed.args.first().ok_or_else(|| {
-                                            syn::Error::new_spanned(
-                                                bracketed,
-                                                "where's the generic args?",
-                                            )
-                                        })? {
-                                            GenericArgument::Type(Type::Path(this_path)) => {
-                                                segments = this_path.path.segments.clone();
-                                                saw_option_ident = true;
-                                                found_option = true;
-                                                continue 'outer;
-                                            }
-                                            _ => continue,
-                                        };
-                                    }
-                                    _ => continue,
-                                },
-                                "SetOfIterator" => saw_setof_iterator = true,
-                                "TableIterator" => {
-                                    if found_option {
-                                        segments =
-                                            Punctuated::from_iter(std::iter::once(segment.clone()));
-                                        found_option = false;
-                                        continue 'outer;
-                                    }
```

Comment:
Yes, this code was completely unnecessary, and the only thing that ever even hits it is one of the functions in the srf example crate! Not even spi_srf!